logic/src/classes/Ride.java

```diff
@@ -1,129 +1,92 @@
-
-//        private Ride originalRide;
-        private Station start;
-        private Station end;
-        List<PartOfRide> relevantPartsOfRide;
-
-        public SubRide(Station start, Station end) {
-//            this.originalRide = originalRide;
-            this.start = start;
-            this.end = end;
-            addRelevantPartsOfRide(start, end);
-        }
-
-        private void addRelevantPartsOfRide(Station start, Station end){
-            //TODO: change to new implementation
-            int indexOfStart = allStations.indexOf(start);
-            int indexOfEnd = allStations.indexOf(end);
-
-            List<Station> subStations = allStations.subList(indexOfStart, indexOfEnd);  //excluding last one
-            relevantPartsOfRide = subStations.stream().map((s) -> station2Part.get(s)).collect(Collectors.toList());
-
-//            relevantPartsOfRide = parts.subList(indexOfStart, indexOfEnd);
-//            PartOfRide current =  parts.stream()
-//                    .filter((p) -> p.getRoad().getStartStation() == start)
-//                    .findFirst()
-//                    .get();
-//            relevantPartsOfRide.add(current);
-//
-//            parts.stream().
-
-        }
+    User rider;
+    Map<Station, ArrayList<Station>> LinkedStation;
+    LinkedList<PartOfRide> path;
+    Schedule schedule;
+
+    public Ride()
+    {
+        rider = new User();
+        path  = new LinkedList<PartOfRide>();
+        LinkedStation = new HashMap<Station, ArrayList<Station>>();
     }
 
-    private static int unique_id = 6000;
-    private final int id;
-    private final User rideOwner;
-//    private List<PartOfRide> parts;
-//    private List<Station> allStations;
-    private HashMap<Station, PartOfRide> station2Part;
-    private Schedule schedule;
-    private int carCapacity;
-
-    private Ride(User rideOwner, HashMap<Station, PartOfRide> station2Part, int carCapacity) {
-        this.id = unique_id++;
-        this.rideOwner=rideOwner;
-        this.carCapacity = carCapacity;
-//        this.parts = partsOfRide;
-        this.schedule = Schedule.SINGLE_TIME;
-        this.station2Part = station2Part;
-
-//        initStationsList(partsOfRide);
+    public void addNewPath(PartOfRide newPath)
+    {
+        LinkedStation.putIfAbsent(newPath.getRoad().getStartStation(), new ArrayList<Station>());
+        LinkedStation.get(newPath.getRoad().getStartStation()).add(newPath.getRoad().getEndStation());
+        path.add(newPath);
     }
 
-    public Ride createRideFromRoads(User rideOwner, List<Road> roads, int capacity){
-        HashMap<Station, PartOfRide> station2Part = new LinkedHashMap<>();
-        roads.forEach((road -> station2Part.put(road.getStartStation(), new PartOfRide(road, capacity))));
-//        List<PartOfRide> partsOfRide = roads.stream().map((r) -> new PartOfRide(r, capacity)).collect(Collectors.toList());
+    public void addTrempist(Trempist trempist, Road specificRoad , SubRide subRide )
```

Comment:
why do you need here both specificRoad and subRide?
SubRide should include all the station already - why I need to point to a specific Road? if I have subRides ==> I have the stations and roads ==> why to send only one every time? just send them all!
Why some one should be aware of subRide Object? this should be internal implementation - no one should be aware of this
What if the provided road is not in this Ride? what if subRide does not contain this stations which in the Road?